src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.shoppingcart.CartItemDto;
+import com.example.onlinebookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.onlinebookstore.exceptions.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto getShoppingCart() {
+        String email = getUserEmail();
+        User user = getUser(email);
+        return shoppingCartMapper.toResponseCart(findCart(user));
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartResponseDto addCartItem(CartItemDto cartItemDto) {
+        String email = getUserEmail();
+        User user = getUser(email);
+        Book book = bookRepository.findById(cartItemDto.bookId()).orElseThrow(
+                () -> new EntityNotFoundException("Non existent book id"));
+        ShoppingCart cart = findCart(user);
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(book);
+        cartItem.setQuantity(cartItemDto.quantity());
+        cartItem.setShoppingCart(cart);
+        cartItemRepository.save(cartItem);
+        return shoppingCartMapper.toResponseCart(cart);
+    }
+
+    @Transactional
```

Comment:
Maybe this method will work without @Transactional. i had to use it only on update method.